src/ui/categories/common/CategoryTreeSelector.js

```diff
@@ -134,19 +73,29 @@ class CategoryTreeSelector extends Component {
                     className="fa fa-minus-square-o CategoryTreeSelector__action-icon"
                     aria-hidden="true"
                   />
-                  Collapse All
+                  <FormattedMessage id="cms.categories.collapseAll" />
                 </span>
               </th>
               <th className="text-center" style={{ width: '4%' }}>
-                Join
+                <FormattedMessage id="cms.join" />
               </th>
             </tr>
           </thead>
-          <tbody>{this.renderRows()}</tbody>
+          <tbody>
+            <CategoryTreeSelectorRow
+              language={language}
+              categories={categories}
+              onExpandCategory={onExpandCategory}
+              onJoinCategory={onJoinCategory}
+              onUnjoinCategory={onUnjoinCategory}
+              input={input}
+            />
+          </tbody>
         </table>
         <br />
         <div>
-          {joinedCategories && joinedCategories.length > 0 ? <h4>Content Category List</h4> : null}
+          {joinedCategories && joinedCategories.length > 0
```

Comment:
as a rule of thumb lets avoid ternary operators inside the return of the render, have the result assigned to a const inside the render and have that just being rendered as it is inside the return.